server/Services/GameRound.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using Api.Hubs;
+using Api.Extensions;
+using System.Threading;
+using Microsoft.AspNetCore.SignalR;
+using Api.DTO;
+
+namespace Api.Services
+{
+  public class GameRound
+  {
+    private readonly int EggCards = 10;
+    private readonly int HenCards = 5;
+    private readonly int FoxCards = 5;
+
+    private Timer GameTimer;
+
+    private List<CardType> Deck { get; set; }
+    private List<CardType> Pile { get; set; }
+
+    private bool CorrectCount(int count) { return count == 5; }
+
+    private IHubContext<GameHub> Hub { get; set; }
+
+    private int GameId { get; set; }
+
+    public GameRound(IHubContext<GameHub> hub, int gameId, TimeSpan cardSpeed)
+    {
+      Hub = hub;
+      GameId = gameId;
+
+      Hub.Clients.Group($"game-{gameId}").SendAsync("roundStart");
+
+      Deck = new List<CardType>();
+      Pile = new List<CardType>();
+
+      for (int i = 0; i < EggCards; i++)
+        Deck.Add(CardType.EGG);
+
+      for (int i = 0; i < HenCards; i++)
+        Deck.Add(CardType.HEN);
+
+      for (int i = 0; i < FoxCards; i++)
+        Deck.Add(CardType.FOX);
+
+      Deck.Shuffle();
+
+      int delay = Convert.ToInt32(cardSpeed.TotalMilliseconds);
+      GameTimer = new Timer(state => DrawCard(), null, delay, delay);
+    }
+
+    private void DrawCard()
+    {
+      if (Deck.Count <= 0)
+      {
+        EndRound(null);
+        return;
+      }
+
+      var card = Deck[0];
+      Deck.RemoveAt(0);
+      Pile.Add(card);
+
+      Hub.Clients.Group($"game-{GameId}").SendAsync("receiveCard", card.ToString());
+    }
+
+    public void EndRound(string playerId)
+    {
+      GameTimer.Dispose();
+      // amount of eggs taking into account hens and foxes
+      int count = 0;
+      int activeChickens = 0;
```

Comment:
Are them chickens of hens? 🐔

Reply:
Hens 😄

Reply:
Then we should rename chickens into hens!

Reply:
Done

Reply:
You forgot `activeChickens`.